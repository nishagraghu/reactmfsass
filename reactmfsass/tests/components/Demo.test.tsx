import renderer from 'react-test-renderer';

import Demo from '../../src/components/Demo';
import React from 'react';
import { render, screen } from "@testing-library/react";

describe("test app", () => {
it("should render the App component", () => {
    const component = renderer.create(
        <Demo />,
      );
      let tree = component.toJSON();
    expect(tree).toMatchSnapshot();


    
});
it('changes the text ', () => {
  
    render(<Demo />);

    const nameElement = screen.getByText("Name: demoreactsass");
    const frameworkElement = screen.getByText("Framework: react");
    const languageElement = screen.getByText("Language: TypeScript");
    const cssElement = screen.getByText("CSS: Empty CSS");

    expect(nameElement).toBeInTheDocument();
    expect(frameworkElement).toBeInTheDocument();
    expect(languageElement).toBeInTheDocument();
    expect(cssElement).toBeInTheDocument();

})
})

