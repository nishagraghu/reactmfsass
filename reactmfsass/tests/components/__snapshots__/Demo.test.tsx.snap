// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app should render the App component 1`] = `
<div
  className="container"
>
  <div>
    Name: demoreactsass
  </div>
  <div>
    Framework: react
  </div>
  <div>
    Language: TypeScript
  </div>
  <div>
    CSS: Empty CSS
  </div>
</div>
`;
